### PR TITLE
Rock64 support

### PR DESCRIPTION
**Pull request by CameronNemo**
~~Needs a void-packages PR for the base package.~~ https://github.com/void-linux/void-packages/pull/27942

### CODE REVIEW

**Comment by jcgruenhage** [comment]
ooi: Is this only pending testing? Or are there still any open questions here? I'm currently building an image based on this and will test it out soon. I figure you've already done that though, so not sure how much help this is.

**Comment by CameronNemo** [comment]
@jcgruenhage there was something funked about the first draft of this PR (bootloader config not getting populated). I successfully tested the agetty's on the console and via HDMI with this new draft.

Some things which I suspect will definitely be broken:

* USB3 port
* HDMI sound
* maybe also the analog audio jack... I never tested that one

**Comment by jcgruenhage** [comment]
I tried this with a mid-2017 Rock64 2GB earlier today, didn't boot from what it looks like. No response to IPv6 link local broadcast pings and no HDMI signal. Will hook up uart later and see whether anything is in the logs.

**Comment by jcgruenhage** [comment]
Okay, so not sure what happened there last time, maybe I just wasn't patient enough, because it does come up now without issues. It does take some time, and the HDMI only comes up once the login screen is ready, but everything seems to work \o/

**Review by jcgruenhage — APPROVED** [review]

**Comment by the-maldridge** [comment]
All editor modelines need to be removed from this PR.

**Comment by CameronNemo** [comment]
dropped that commit

**Comment by Anachron** [comment]
Does this support `ROCKPro64` as well or only the `Rock64` models?

**Comment by jcgruenhage** [comment]
@Anachron this branch is only for adding Rock64 support, but the RockPro64 is already supported by void-mklive as it is, so you can just take the `master` branch for that.

**Comment by CameronNemo** [comment]
rebased

File: mkimage.sh.in [hunk 1]
```diff
@@ -291,9 +295,32 @@ ci20*)
...
+    mkdir -p "${ROOTFS}/boot/extlinux"
+    run_cmd_chroot "${ROOTFS}" "/etc/kernel.d/post-install/60-extlinux"
+    # configure and enable services
+    cat >"${ROOTFS}/etc/sv/agetty-console/conf" <<_EOF
+BAUD_RATE=1500000
+TERM_NAME=linux
+_EOF
+    for _service in dhcpcd sshd agetty-console ; do
+        run_cmd_chroot "${ROOTFS}" "ln -sv /etc/sv/$_service /etc/runit/runsvdir/default/$_service"
+    done
```
Comment (Johnnynator) [inline mkimage.sh.in:319]: I think we should put this into it's own function, and maybe do this in a general fashion, instead of using `INSTALL` scripts in the other `-base` packages. Can you do this as a separate PR for discussion?

Comment (CameronNemo) [reply mkimage.sh.in:319]: hrm... wouldn't the image be unusable without these services activated?

Comment (Johnnynator) [reply mkimage.sh.in:319]: I mean, we should either do it `INSTALL` for all boards, or change it to be done in `mklive` for all of them. Not mixing it.

I think my personal preference would be to do the configuration of initial services for such boards in `mklive`

**Comment by CameronNemo** [comment]
Rebased and removed service configuration, but not yet tested.

**Comment by CameronNemo** [comment]
Tested successfully (both PBP and rock64).

**Comment by Idesmi** [comment]
Currently Void doesn't boot on Rock64.
I can upload logs if there are any and I am adviced what to provide.

**Comment by CameronNemo** [comment]
Do you have a serial adapter? What does "not boot" mean? What point in the boot stage does it get to (ROM, U-Boot, Linux, initramfs, Void rootfs)?

**Comment by jcgruenhage** [comment]
ftr, I do also still have weird problems with Void on my Rock64 SBCs. They sometimes just don't come up, and I'm tired of trying, thinking about just buying other hardware and abandoning them..

I've tried figuring out where it fails, but wasn't successful, because each time I had a serial cable connected with it on my desk, it came up successfully, but when sitting on my rack with otherwise identical connectivity, it regularly doesn't boot, and I can't figure out why. I haven't had this problem with other distros either, armbian boots reliably for example.

If you people have any ideas how I might debug this heisenbug, I'm all ears.

**Comment by leahneukirchen** [comment]
Using the netconsole may be an option to get kernel logs when it fails: https://www.kernel.org/doc/Documentation/networking/netconsole.txt

Alternatively, you can try dumping it to the sdcard: https://www.kernel.org/doc/html/latest/admin-guide/pstore-blk.html

**Comment by Idesmi** [comment]
@CameronNemo I don't have a serial adapter. I am unable to see any live output through HDMI; no signal is sent, while for example Armbian sends signal as soon as the board is powered on.

Unless you can explain how to log the boot process in a different way, I can only tell that it doesn't get as far as to register an IP on the network.
